src/Compilers/Core/Portable/Text/SourceText.cs

```diff
@@ -415,14 +415,11 @@ public virtual void Write(TextWriter writer, TextSpan span, CancellationToken ca
             }
         }
 
-        internal ImmutableArray<byte> GetChecksum(bool useDefaultEncodingIfNull = false)
+        public ImmutableArray<byte> GetChecksum()
         {
             if (_lazyChecksum.IsDefault)
             {
-                // we shouldn't be asking for a checksum of encoding-less source text, except for SourceText comparison.
-                Debug.Assert(this.Encoding != null || useDefaultEncodingIfNull);
-
-                using (var stream = new SourceTextStream(this, useDefaultEncodingIfNull: useDefaultEncodingIfNull))
+                using (var stream = new SourceTextStream(this, useDefaultEncodingIfNull: true))
```

Comment:
The default value of the parameter was `false` but you're explicitly passing `true` here.  This should be `false` to match the old behavior.